vispy/visuals/image.py

```diff
@@ -67,15 +68,33 @@
 
 _null_color_transform = 'vec4 pass(vec4 color) { return color; }'
 _c2l = 'float cmap(vec4 color) { return (color.r + color.g + color.b) / 3.; }'
+
+_ipol_template = """
+    #include "misc/spatial-filters.frag"
+    vec4 texture_lookup(vec2 texcoord) {
+        /*if(texcoord.x < 0.0 || texcoord.x > 1.0 ||
+        texcoord.y < 0.0 || texcoord.y > 1.0) {
+            discard;
+        }*/
+        return %s($texture, $shape, texcoord);
+    }"""
+
 _texture_lookup = """
-    vec4 texture_lookup(vec2 texcoord) { 
+    vec4 texture_lookup(vec2 texcoord) {
         if(texcoord.x < 0.0 || texcoord.x > 1.0 ||
         texcoord.y < 0.0 || texcoord.y > 1.0) {
             discard;
         }
-        return texture2D($texture, texcoord); 
+        return texture2D($texture, texcoord);
     }"""
 
+def create_ipol_frag():
+    from ..io import load_spatial_filters
+    from ..ext.ordereddict import OrderedDict
```

Comment:
Ah, yes, I wasn't sure if that function will stay this way, so I have it there for the time being. This will go into import section when finalizing. Thanks for catching.